src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -613,19 +609,21 @@ public DisplayVersion SelectedVersion
                         {
                             PackageMetadata = detailedPackageMetadata;
                         }
-
-                        NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                        else
```

Comment:
For my understanding, can you please explain what the behavior difference is after adding this else block?

Reply:
The if block before is checking to see if the metadata is available in the _metadataDict dictionary. So without the else we would say we found it in the dictionary then look for it again.